agent/consul/server_serf.go

```diff
@@ -37,7 +37,9 @@ func (s *Server) setupSerf(conf *serf.Config, ch chan serf.Event, path string, w
 		conf.NodeName = fmt.Sprintf("%s.%s", s.config.NodeName, s.config.Datacenter)
 	} else {
 		conf.NodeName = s.config.NodeName
-		conf.Tags["wan_join_port"] = fmt.Sprintf("%d", wanPort)
+		if wanPort > 0 {
+			conf.Tags["wan_join_port"] = fmt.Sprintf("%d", wanPort)
```

Comment:
For the record, I checked and it seems the only consumer of that tag in our code already handles the case where it's not present: https://github.com/hashicorp/consul/blob/7de57ba4dea80b0fa02e517d673b0d727d2b5a3b/agent/metadata/server.go#L119-L125